build.rs

```diff
@@ -14,9 +14,86 @@
 // You should have received a copy of the GNU General Public License
 // along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
 
+//! Build script to ensure that node/worker versions stay in sync and PVF worker binaries are always
+//! built whenever polkadot is built.
+
+use std::{env::var, path::Path, process::Command};
+
+// TODO: fix
+// use polkadot_node_core_pvf::{PREPARE_BINARY_NAME, EXECUTE_BINARY_NAME};
+const PREPARE_BINARY_NAME: &str = "polkadot-prepare-worker";
+const EXECUTE_BINARY_NAME: &str = "polkadot-execute-worker";
+
 fn main() {
+	// Always build PVF worker binaries whenever polkadot is built.
+	//
+	// This is needed because `default-members` does the same thing, but only for `cargo build` --
+	// it does not work for `cargo run`.
+	//
+	// TODO: Test with `cargo +<specific-toolchain> ...`
+	{
+		let cargo = var("CARGO").expect("`CARGO` env variable is always set by cargo");
+		let target = var("TARGET").expect("`TARGET` env variable is always set by cargo");
+		let profile = var("PROFILE").expect("`PROFILE` env variable is always set by cargo");
+		let out_dir = var("OUT_DIR").expect("`OUT_DIR` env variable is always set by cargo");
+		let target_dir = format!("{}/workers", out_dir);
+
+		// Settings like overflow-checks, opt-level, lto, etc. are correctly passed to cargo
+		// subcommand through env vars, e.g. `CARGO_CFG_OVERFLOW_CHECKS`, which the subcommand
+		// inherits. We don't pass along features because the workers don't use any right now.
+		let mut args = vec![
+			"build",
+			"-p",
+			EXECUTE_BINARY_NAME,
+			"-p",
+			PREPARE_BINARY_NAME,
+			"--target",
+			&target,
+			"--target-dir",
+			&target_dir,
+		];
+		if profile != "debug" {
+			args.push("--profile");
+			args.push(&profile);
+		}
+		let mut build_cmd = Command::new(cargo);
+		build_cmd.args(&args);
+
+		println!(
```

Comment:
I believe it should be `eprintln!()` as stdout is used by build scripts to communicate with Cargo

Reply:
I got it from [wasm_project](https://github.com/paritytech/substrate/blob/aedfdc977b/utils/wasm-builder/src/wasm_project.rs#L696), will try `eprintln!`.